lib/src/chewie_controller.dart

```diff
@@ -0,0 +1,165 @@
+  /// [CupertinoControls] for reference.
+  final Widget customControls;
+
+  /// The Aspect Ratio of the Video. Important to get the correct size of the
+  /// video!
+  ///
+  /// Will fallback to fitting within the space allowed.
+  final double aspectRatio;
+
+  /// The colors to use for controls on iOS. By default, the iOS player uses
+  /// colors sampled from the original iOS 11 designs.
+  final ChewieProgressColors cupertinoProgressColors;
+
+  /// The colors to use for the Material Progress Bar. By default, the Material
+  /// player uses the colors from your Theme.
+  final ChewieProgressColors materialProgressColors;
+
+  /// The placeholder is displayed underneath the Video before it is initialized
+  /// or played.
+  final Widget placeholder;
+
+  /// Defines if the player will start in fullscreen when play is pressed
+  final bool fullScreenByDefault;
+
+  /// Defines if the player will sleep in fullscreen or not
+  final bool allowedScreenSleep;
+
+  /// Defines if the controls should be for live stream video
+  final bool isLive;
+
+  bool _isFullScreen = false;
+
+  bool get isFullScreen => _isFullScreen;
+
+  Future _initialize() async {
+    await videoPlayerController.setLooping(looping);
+
+    if ((autoInitialize || autoPlay) &&
+        !videoPlayerController.value.initialized) {
+      await videoPlayerController.initialize();
+    }
+
+    if (autoPlay) {
+      if (fullScreenByDefault) {
+        enterFullscreen();
+      }
+
+      await videoPlayerController.play();
+    }
+
+    if (startAt != null) {
+      await videoPlayerController.seekTo(startAt);
+    }
+
+    if (fullScreenByDefault) {
+      videoPlayerController.addListener(() async {
+        if (await videoPlayerController.value.isPlaying && !_isFullScreen) {
+          enterFullscreen();
+        }
+      });
+    }
+  }
+
+  void enterFullscreen() {
+    _isFullScreen = true;
+    notifyListeners();
+  }
+
+  void exitFullscreen() {
+    _isFullScreen = false;
+    notifyListeners();
+  }
+
+  void toggleFullscreen() {
+    _isFullScreen = !_isFullScreen;
+    notifyListeners();
+  }
+
+  Future<void> play() async {
+    await videoPlayerController.play();
+  }
+
+  Future<void> setLooping(bool looping) async {
+    await videoPlayerController.setLooping(looping);
+  }
+
+  Future<void> pause() async {
+    await videoPlayerController.pause();
+  }
+
+  Future<void> seekTo(Duration moment) async {
+    await videoPlayerController.seekTo(moment);
+  }
+
+  Future<void> setVolume(double volume) async {
+    await videoPlayerController.setVolume(volume);
+  }
+}
+
+class ChewieControllerProvider extends InheritedWidget {
+  const ChewieControllerProvider({
+    Key key,
+    @required this.controller,
+    @required Widget child,
+  })  : assert(controller != null),
+        assert(child != null),
+        super(key: key, child: child);
+
+  final ChewieController controller;
+
+  static ChewieController of(BuildContext context) {
```

Comment:
Can we move this method to the `ChewieController` class? That will make it so you can access the ChewieController via `ChewieController.of` instead of `ChewieControllerProvider.of`. This matches the same style as other "Inherited" data, such as `Theme.of`.

We could even make this provider private! Such as `_ChewieControllerProvider` and use it only internally.

Reply:
`ChewieController.of` is indeed much nicer. I somehow missed that this could be done in this way.